# SHORT LISTS OF CARM_VIS OBJECTS

This small notebook just gets and stores short lists of the CARM_VIS objets, making two subsamples:

- The first subsample chooses those objects that have correctly calculated GLS periodograms (for both RV and TESS curves), that will be the input to the periodogram / stacked periodogram analysis. 
- The second subsample, choose the RV curves with low RMS values (no matter if they have the GLS periodograms correctly calculated or not).


## Modules and configuration

### Modules

In [1]:
import pandas as pd
import numpy as np

### Configuration

In [2]:
INPUT_FILE = "../data/CARM_VIS_objects_with_PG.csv"
OUTPUT_FILE_PG = "../data/SELECTION_for_PG_CARM_VIS_objects_with_PG.csv"
OUTPUT_FILE_ML = "../data/SELECTION_for_ML_CARM_VIS_objects_with_PG.csv"

MAX_RMS = 5.0 #  Maximum value allowed for RMS to select the record as input to ML analysis.

##  Load the CARM_VIS objects

In [3]:
df = pd.read_csv(INPUT_FILE, sep=',', decimal='.')
df.head()

Karmn             Name Comp    GJ  RA_J2016_deg  DE_J2016_deg  \
0  J23585+076        Wolf 1051   AB  4383    359.636420      7.656947   
1  J23556-061           GJ 912   AB   912    358.913617     -6.144283   
2  J23548+385  RX J2354.8+3831    -   NaN    358.713658     38.526340   
3  J23505-095       LP 763-012    -  4367    357.634705     -9.560964   
4  J23492+024           BR Psc    -   908    357.306604      2.396918   

      RA_J2000     DE_J2000  l_J2016_deg  b_J2016_deg  ... WF_T0_PG_TESS  \
0  23:58:32.65  +07:39:30.1   100.839686   -52.931049  ...           NaN   
1  23:55:39.78  -06:08:33.4    88.129933   -65.175491  ...           NaN   
2  23:54:51.46  +38:31:36.2   110.941908   -23.024449  ...   1764.689208   
3  23:50:31.64  -09:33:32.7    80.777067   -67.303426  ...   1353.926624   
4  23:49:12.53  +02:24:04.4    93.567467   -56.885396  ...           NaN   

  WF_e_T0_PG_TESS  WF_offset_PG_TESS WF_e_offset_PG_TESS  WF_FAP_PG_TESS  \
0             NaN                NaN                 NaN             NaN   
1             NaN                NaN                 NaN             NaN   
2        0.027575         999.999756        2.151008e-06             1.0   
3        0.036481        1000.000122        9.022946e-07             1.0   
4             NaN                NaN                 NaN             NaN   

   WF_valid_PG_TESS                                   WF_error_PG_TESS  \
0               0.0  Not recognized as a supported data product:\nn...   
1               0.0  Not recognized as a supported data product:\nn...   
2               1.0                                                NaN   
3               1.0                                                NaN   
4               0.0  Not recognized as a supported data product:\nn...   

   WF_elapsed_time_PG_TESS                                 WF_plain_file_TESS  \
0                 0.000997                                                NaN   
1                 0.000998                                                NaN   
2                94.758838  ../data/CARM_VIS_TESS_WinFunc_PGs/WF_J23548+38...   
3               132.607176  ../data/CARM_VIS_TESS_WinFunc_PGs/WF_J23505-09...   
4                 0.001995                                                NaN   

                                    WF_fig_file_TESS  
0                                                NaN  
1                                                NaN  
2  ../data/CARM_VIS_TESS_WinFunc_PGs/figures/WF_J...  
3  ../data/CARM_VIS_TESS_WinFunc_PGs/figures/WF_J...  
4                                                NaN  

[5 rows x 298 columns]

In [4]:
print(list(df.columns))

['Karmn', 'Name', 'Comp', 'GJ', 'RA_J2016_deg', 'DE_J2016_deg', 'RA_J2000', 'DE_J2000', 'l_J2016_deg', 'b_J2016_deg', 'Ref01', 'SpT', 'SpTnum', 'Ref02', 'Teff_K', 'eTeff_K', 'logg', 'elogg', '[Fe/H]', 'e[Fe/H]', 'Ref03', 'L_Lsol', 'eL_Lsol', 'Ref04', 'R_Rsol', 'eR_Rsol', 'Ref05', 'M_Msol', 'eM_Msol', 'Ref06', 'muRA_masa-1', 'emuRA_masa-1', 'muDE_masa-1', 'emuDE_masa-1', 'Ref07', 'pi_mas', 'epi_mas', 'Ref08', 'd_pc', 'ed_pc', 'Ref09', 'Vr_kms-1', 'eVr_kms-1', 'Ref10', 'ruwe', 'Ref11', 'U_kms-1', 'eU_kms-1', 'V_kms-1', 'eV_kms-1', 'W_kms-1', 'eW_kms-1', 'Ref12', 'sa_m/s/a', 'esa_m/s/a', 'Ref13', 'SKG', 'Ref14', 'SKG_lit', 'Ref14_lit', 'Pop', 'Ref15', 'vsini_flag', 'vsini_kms-1', 'evsini_kms-1', 'Ref16', 'P_d', 'eP_d', 'Ref17', 'pEWHalpha_A', 'epEWHalpha_A', 'Ref18', 'log(LHalpha/Lbol)', 'elog(LHalpha/Lbol)', 'Ref19', '1RXS', 'CRT_s-1', 'eCRT_s-1', 'HR1', 'eHR1', 'HR2', 'eHR2', 'Flux_X_E-13_ergcm-2s-1', 'eFlux_X_E-13_ergcm-2s-1', 'LX/LJ', 'eLX/LJ', 'Ref20', 'Activity', 'Ref21', 'FUV_mag',

In [5]:
df['lc_file']

0                                                    NaN
1                                                    NaN
2      ../data/CARMENES_GTO_TESS_lc/lightcurves/J2354...
3      ../data/CARMENES_GTO_TESS_lc/lightcurves/J2350...
4                                                    NaN
                             ...                        
385    ../data/CARMENES_GTO_TESS_lc/lightcurves/J0018...
386                                                  NaN
387                                                  NaN
388                                                  NaN
389    ../data/CARMENES_GTO_TESS_lc/lightcurves/J0005...
Name: lc_file, Length: 390, dtype: object

Number of CARM_VIS objects without an RV file name:

In [6]:
sum(df['rv_file'].isna())

0

Number of CARM_VIS objects without an LC file name:

In [7]:
sum(df['lc_file'].isna())

119

## Select the CARM_VIS object with valid PG calculated

Total number of CARM_VIS objects in the original table:

In [8]:
len(df)

390

Number of CARM_VIS objects with correctly calculated periodogram for CARMENES RV curves:

In [9]:
len(df[df['valid_PG_RV'] == 1.0])

387

Number of CARM_VIS objects with correctly calculated periodogram for TESS LC:

In [10]:
len(df[df['valid_PG_TESS'] == 1.0])

267

Combined number of CARM_VIS objects with correctly calculated periodograms for both CARMENES RV curves and TESS LC:

In [11]:
len(df[(df['valid_PG_RV'] == 1.0) & (df['valid_PG_TESS'] == 1.0)])

265

These 265 objects will be the main target of the periodogram analysis:

In [12]:
selected_df = df[(df['valid_PG_RV'] == 1.0) & (df['valid_PG_TESS'] == 1.0)].copy()
selected_df.head()

Karmn             Name Comp    GJ  RA_J2016_deg  DE_J2016_deg  \
2   J23548+385  RX J2354.8+3831    -   NaN    358.713658     38.526340   
3   J23505-095       LP 763-012    -  4367    357.634705     -9.560964   
5   J23431+365          GJ 1289    -  1289    355.781509     36.536310   
7   J23381-162        G 273-093    -  4352    354.532687    -16.236514   
10  J23245+578       BD+57 2735    -   895    351.126628     57.853057   

       RA_J2000     DE_J2000  l_J2016_deg  b_J2016_deg  ... WF_T0_PG_TESS  \
2   23:54:51.46  +38:31:36.2   110.941908   -23.024449  ...   1764.689208   
3   23:50:31.64  -09:33:32.7    80.777067   -67.303426  ...   1353.926624   
5   23:43:06.31  +36:32:13.1   107.922839   -24.336479  ...   1764.702841   
7   23:38:08.16  -16:14:10.2    61.845437   -69.825220  ...   1353.926631   
10  23:24:30.51  +57:51:15.5   111.552287    -3.085183  ...   1955.851155   

   WF_e_T0_PG_TESS  WF_offset_PG_TESS WF_e_offset_PG_TESS  WF_FAP_PG_TESS  \
2         0.027575         999.999756        2.151008e-06             1.0   
3         0.036481        1000.000122        9.022946e-07             1.0   
5         0.028355         999.999512        4.306074e-06             1.0   
7         0.036480        1000.000122        9.022946e-07             1.0   
10        0.018437         999.999512        3.720858e-06             1.0   

    WF_valid_PG_TESS  WF_error_PG_TESS  WF_elapsed_time_PG_TESS  \
2                1.0               NaN                94.758838   
3                1.0               NaN               132.607176   
5                1.0               NaN                97.939914   
7                1.0               NaN               136.603404   
10               1.0               NaN               131.327304   

                                   WF_plain_file_TESS  \
2   ../data/CARM_VIS_TESS_WinFunc_PGs/WF_J23548+38...   
3   ../data/CARM_VIS_TESS_WinFunc_PGs/WF_J23505-09...   
5   ../data/CARM_VIS_TESS_WinFunc_PGs/WF_J23431+36...   
7   ../data/CARM_VIS_TESS_WinFunc_PGs/WF_J23381-16...   
10  ../data/CARM_VIS_TESS_WinFunc_PGs/WF_J23245+57...   

                                     WF_fig_file_TESS  
2   ../data/CARM_VIS_TESS_WinFunc_PGs/figures/WF_J...  
3   ../data/CARM_VIS_TESS_WinFunc_PGs/figures/WF_J...  
5   ../data/CARM_VIS_TESS_WinFunc_PGs/figures/WF_J...  
7   ../data/CARM_VIS_TESS_WinFunc_PGs/figures/WF_J...  
10  ../data/CARM_VIS_TESS_WinFunc_PGs/figures/WF_J...  

[5 rows x 298 columns]

In [13]:
len(selected_df)

265

### Store the filtered results

In [14]:
selected_df.to_csv(OUTPUT_FILE_PG, sep=',', decimal='.', index=False)

## Select the CARM_VIS objects with lower RMS values

Total number of CARM_VIS objects in the original table:

In [15]:
len(df)

390

RMS values statistics:

In [16]:
df['wrms_RV'].describe()

count      387.000000
mean       361.871528
std       2564.902866
min          0.666710
25%          2.684700
50%          4.015005
75%         11.644566
max      31233.301918
Name: wrms_RV, dtype: float64

In [17]:
len(df[df['wrms_RV'] < MAX_RMS])

233

These 233 objects will be the main target for the Machine Learning approach:

In [19]:
selected_df = df[df['wrms_RV'] < MAX_RMS].copy()
selected_df.head()

Karmn        Name Comp    GJ  RA_J2016_deg  DE_J2016_deg     RA_J2000  \
3  J23505-095  LP 763-012    -  4367    357.634705     -9.560964  23:50:31.64   
4  J23492+024      BR Psc    -   908    357.306604      2.396918  23:49:12.53   
5  J23431+365     GJ 1289    -  1289    355.781509     36.536310  23:43:06.31   
6  J23419+441      HH And    -   905    355.480015     44.170376  23:41:55.04   
7  J23381-162   G 273-093    -  4352    354.532687    -16.236514  23:38:08.16   

      DE_J2000  l_J2016_deg  b_J2016_deg  ...   T0_PG_TESS e_T0_PG_TESS  \
3  -09:33:32.7    80.777067   -67.303426  ...  1354.001496     0.011641   
4  +02:24:04.4    93.567467   -56.885396  ...          NaN          NaN   
5  +36:32:13.1   107.922839   -24.336479  ...  1764.717379     0.006457   
6  +44:10:38.8   109.989338   -16.947350  ...          NaN          NaN   
7  -16:14:10.2    61.845437   -69.825220  ...  1354.111110     0.000422   

   offset_PG_TESS e_offset_PG_TESS  FAP_PG_TESS  valid_PG_TESS  \
3     7767.134369         0.094293     0.049768            1.0   
4             NaN              NaN          NaN            0.0   
5    16158.285568         0.164666     0.000484            1.0   
6             NaN              NaN          NaN            0.0   
7    30353.147895         0.175123     0.063028            1.0   

                                       error_PG_TESS  elapsed_time_PG_TESS  \
3                                                NaN            133.687905   
4  Not recognized as a supported data product:\nn...              0.000997   
5                                                NaN             99.903255   
6  Not recognized as a supported data product:\nn...              0.001995   
7                                                NaN            137.140038   

                                      fits_file_TESS  \
3  ../data/CARM_VIS_TESS_PGs/J23505-095_TESS_PG.fits   
4                                                NaN   
5  ../data/CARM_VIS_TESS_PGs/J23431+365_TESS_PG.fits   
6                                                NaN   
7  ../data/CARM_VIS_TESS_PGs/J23381-162_TESS_PG.fits   

                                       fig_file_TESS  
3  ../data/CARM_VIS_TESS_PGs/figures/J23505-095_T...  
4                                                NaN  
5  ../data/CARM_VIS_TESS_PGs/figures/J23431+365_T...  
6                                                NaN  
7  ../data/CARM_VIS_TESS_PGs/figures/J23381-162_T...  

[5 rows x 248 columns]

In [21]:
len(selected_df)

233

### Store the filtered results

In [25]:
selected_df.to_csv(OUTPUT_FILE_ML, sep=',', decimal='.', index=False)

**Note:** no matter if RV periodograms were correctly calculated or not, although in this case all the selected objects have correctly calculated RV periodogrms.

In [24]:
selected_df[selected_df['valid_PG_RV'] == 0.0]

Empty DataFrame
Columns: [Karmn, Name, Comp, GJ, RA_J2016_deg, DE_J2016_deg, RA_J2000, DE_J2000, l_J2016_deg, b_J2016_deg, Ref01, SpT, SpTnum, Ref02, Teff_K, eTeff_K, logg, elogg, [Fe/H], e[Fe/H], Ref03, L_Lsol, eL_Lsol, Ref04, R_Rsol, eR_Rsol, Ref05, M_Msol, eM_Msol, Ref06, muRA_masa-1, emuRA_masa-1, muDE_masa-1, emuDE_masa-1, Ref07, pi_mas, epi_mas, Ref08, d_pc, ed_pc, Ref09, Vr_kms-1, eVr_kms-1, Ref10, ruwe, Ref11, U_kms-1, eU_kms-1, V_kms-1, eV_kms-1, W_kms-1, eW_kms-1, Ref12, sa_m/s/a, esa_m/s/a, Ref13, SKG, Ref14, SKG_lit, Ref14_lit, Pop, Ref15, vsini_flag, vsini_kms-1, evsini_kms-1, Ref16, P_d, eP_d, Ref17, pEWHalpha_A, epEWHalpha_A, Ref18, log(LHalpha/Lbol), elog(LHalpha/Lbol), Ref19, 1RXS, CRT_s-1, eCRT_s-1, HR1, eHR1, HR2, eHR2, Flux_X_E-13_ergcm-2s-1, eFlux_X_E-13_ergcm-2s-1, LX/LJ, eLX/LJ, Ref20, Activity, Ref21, FUV_mag, eFUV_mag, NUV_mag, eNUV_mag, Ref22, u_mag, eu_mag, Ref23, BT_mag, eBT_mag, Ref24, ...]
Index: []

[0 rows x 248 columns]

Of these objects, in case we would like to combine periodograms with Machine Learning, we have the following objects with all the information:

In [26]:
len(selected_df[(selected_df['valid_PG_RV'] == 1) & (selected_df['valid_PG_TESS'] == 1)])

161

# Summary

**OBSERVATIONS AND CONCLUSIONS:**
From the 390 objects of the CARMENES NZP-corrected RV time series, we can use:
- 265 objects for periodogram analysis (for them, both RV and _TESS_ periodograms are available).
- 233 objects for Machine Learning analysis (objects with RMS < 5 $ms^{-1}$).
- Maybe we can use just the more limited 161 objects with RMS < 5 $ms^{-1}$ for a combined Periodogram / Machine Learning analysis, as all the relevant information is available for these objects.